### DISCUSSION TITLE
Disabling the service_role

### DISCUSSION THREAD

**Discussion by arikalon1** [supabase/supabase#17202] [Questions]
Hi,

Is it possible to run supabase without the `service_role`
Due to security reasons, on some self hosted installations, we don't want to have such a permissive user which never expires.

For our own applications, we don't use the `service_role`

Thanks

**Comment by steve-chavez** [comment]
Can you try dropping it?

```sql
DROP ROLE service_role;
```

**Reply by arikalon1** [reply]
Thanks @steve-chavez 

I think I can technically drop it, I just want to be sure I'm not breaking any existing internal functionality that might be using it

**Reply by steve-chavez** [reply]
Only  the API docs rely on `anon`, I don't recall other service relying on the `service_role`. 

Hm, the API docs also show the JWT for the `service_role`. In theory it should be possible to not show it if it's deleted, but not sure if it does now.

**Reply by arikalon1** [reply]
Thanks for your reply @steve-chavez
